chia/cmds/cmds_util.py

```diff
@@ -10,3 +27,131 @@ def transaction_submitted_msg(tx: TransactionRecord) -> str:
 
 def transaction_status_msg(fingerprint: int, tx_id: bytes32) -> str:
     return f"Run 'chia wallet get_transaction -f {fingerprint} -tx 0x{tx_id}' to get status"
+
+
+async def validate_client_connection(
+    rpc_client: RpcClient, node_type: str, rpc_port: int, fingerprint: Optional[int]
+) -> Tuple[Optional[RpcClient], Optional[int]]:
+    from aiohttp import ClientConnectorError
+
+    try:
+        await rpc_client.healthz()
+        if type(rpc_client) == WalletRpcClient:
+            wallet_and_f = await get_wallet(rpc_client, fingerprint=fingerprint)
+            if wallet_and_f is None:
+                return None, fingerprint
+            else:
+                rpc_client, fingerprint = wallet_and_f
+    except ClientConnectorError:
+        print(f"Connection error. Check if {node_type.replace('_', ' ')} rpc is running at {rpc_port}")
+        print(f"This is normal if {node_type.replace('_', ' ')} is still starting up")
+        return None, fingerprint
+    return rpc_client, fingerprint
+
+
+@asynccontextmanager
+async def get_any_service_client(
+    node_type: str,
+    rpc_port: Optional[int] = None,
+    root_path: Path = DEFAULT_ROOT_PATH,
+    fingerprint: Optional[int] = None,
+) -> AsyncIterator[Tuple[Optional[Any], Dict[str, Any], Optional[int]]]:
+    # in the return the dict is the config & the first element is the rpc client.
+    from chia.util.config import load_config
+    from chia.util.ints import uint16
+
+    if node_type not in NODE_TYPES.keys():
+        print(f"Invalid node type: {node_type}")
+        return
```

Comment:
> I would put this in a separate function for mapping a string to a node type.
> 
> As is I think it is some form of wrong in terms of not yielding to the body of the context manager and also not raising an exception. Looks like you get a `RuntimeError` when you do this.
> 
> ```python
> from contextlib import contextmanager
> 
> 
> @contextmanager
> def m():
>     return
>     yield
> 
> 
> with m():
>     print("inside")
> ```
> 
> ```
> Traceback (most recent call last):
>   File "main.py", line 10, in <module>
>     with m():
>   File "/nix/store/p21fdyxqb3yqflpim7g8s1mymgpnqiv7-python3-3.8.12/lib/python3.8/contextlib.py", line 115, in __enter__
>     raise RuntimeError("generator didn't yield") from None
> RuntimeError: generator didn't yield
> ```

yeah it was originally an async iterator and i forgot to fix that.
Click filters stuff out so that should never happen.